luarules/gadgets/unit_explosion_spawner.lua

```diff
@@ -67,6 +67,7 @@ local expireCount = 0
 local spawnList = {} -- [index] = {.spawnDef, .teamID, .x, .y, .z, .ownerID}, subtables reused
 local spawnCount = 0
 local spawnNames = {}
+local minWaterDepth = -12 --calibrated off of the armpw's (minimum found) maxwaterdepth value
```

Comment:
If this is supposed to the minimum `maxwaterdepth` in any unit def, that can be found dynamically at runtime, it does not have to be hardcoded.

Reply:
Ran out of time, this is a Band-Aid. 

Accessing it on a per unit basis at run time would require a significant rework of the way the explosion spawner works